### PR TITLE
Correction formulaire

### PR DESCRIPTION
**Pull request by SebastienPeillet**
Several simple fixes on forms :  

- `podSize` range
- `visibility` type
- orange style sheet on `back` and `fishActivity` fields
- avoid duplication of `comment` field
- authorize Null value for `podSize`, `distance`, `angle` fields, to avoid a default value emergence
- `nObserver` value can be set to 0

and two fixes a little bit more complicated:

- on followers table, adding a second row removes the `back` field value of the first one, which has been correctly duplicated in the second. It append during the second row addition. I didn't find the cause of this, I only suppose it is related with Value Relation widget. I added a fix that re-applies original value on the first row. Works, but weird.

- on docked tables, it appears that adding two rows at a time (by using shortcut for instance) make a bug on the first column of the table. Then it's impossible to modify the value of this specific column. I force the focus on the map canvas before a row addition to save the previous one and avoid the bug. Simple, it seems like a workaround made mandatory due to table docking.

### CODE REVIEW

**Comment by pblottiere** [comment]
I tried this branch on master and 3.26 and I still have a segfault as soon as I click on the `Update environment` button:

```
Python Stack Trace
Fatal Python error: Segmentation fault

Thread 0x00007f60c265c6c0 (most recent call first):
  

Thread 0x00007f60c2e5d6c0 (most recent call first):
  File "/usr/lib/python3.10/threading.py", line 320 in wait
    waiter.acquire()
  File "/usr/lib/python3.10/queue.py", line 171 in get
    self.not_empty.wait()
  File "/home/pblottiere/.local/share/QGIS/QGIS3/profiles/light/python/plugins/sammo-boat/src/core/thread_sound_recording.py", line 44 in _recordingLoop
    file.write(self._queue.get())
  File "/home/pblottiere/.local/share/QGIS/QGIS3/profiles/light/python/plugins/sammo-boat/src/core/thread_sound_recording.py", line 41 in _toDoInsideLoop
    self._recordingLoop(file)
  File "/home/pblottiere/.local/share/QGIS/QGIS3/profiles/light/python/plugins/sammo-boat/src/core/other_thread.py", line 23 in run
    self._toDoInsideLoop()

Thread 0x00007f60cf7fe6c0 (most recent call first):
  File "/home/pblottiere/.local/share/QGIS/QGIS3/profiles/light/python/plugins/sammo-boat/src/core/thread_widget.py", line 19 in _toDoInsideLoop
    sleep(2)
  File "/home/pblottiere/.local/share/QGIS/QGIS3/profiles/light/python/plugins/sammo-boat/src/core/other_thread.py", line 23 in run
    self._toDoInsideLoop()

Thread 0x00007f60cffff6c0 (most recent call first):
  File "/home/pblottiere/.local/share/QGIS/QGIS3/profiles/light/python/plugins/sammo-boat/src/core/gps.py", line 47 in _toDoInsideLoop
    time.sleep(1.0)
  File "/home/pblottiere/.local/share/QGIS/QGIS3/profiles/light/python/plugins/sammo-boat/src/core/other_thread.py", line 23 in run
    self._toDoInsideLoop()

Current thread 0x00007f61b6f8bd00 (most recent call first):
  File "/home/pblottiere/.local/share/QGIS/QGIS3/profiles/light/python/plugins/sammo-boat/src/gui/attribute_table.py", line 33 in refresh
    view.setColumnWidth(
  File "/home/pblottiere/.local/share/QGIS/QGIS3/profiles/light/python/plugins/sammo-boat/src/gui/table.py", line 112 in refresh
    SammoAttributeTable.refresh(table, layer.name())
  File "/home/pblottiere/.local/share/QGIS/QGIS3/profiles/light/python/plugins/sammo-boat/sammo.py", line 422 in onEnvironmentAction
    self.tableDock.refresh(layer)
  File "/home/pblottiere/.local/share/QGIS/QGIS3/profiles/light/python/plugins/sammo-boat/src/gui/environment.py", line 38 in onClick
    self.updateEnvironment.emit(self.action.isChecked()) 
```

**Comment by SebastienPeillet** [comment]
Hmmm weird, I tested it on windows also this morning, with a brand new install and everything works fine :/

**Comment by pblottiere** [comment]
I also have some errors in the console when I click on the `update environment` button:

``` console
ERROR 1: In ExecuteSQL(): sqlite3_step(SELECT COUNT(*) FROM (SELECT 1 FROM "plateform" LIMIT 100000)):
  unable to open database file
ERROR 1: sqlite3_get_table(SELECT feature_count FROM gpkg_ogr_contents WHERE lower(table_name) = lower('plateform') LIMIT 2) failed: unable to open database file
```

> Hmmm weird, I tested it on windows also this morning, with a brand new install and everything works fine :/

It works sometimes, so let's say that my environment is buggy

**Comment by pblottiere** [comment]
Mmmm... I don' have segfaults on master nor the error messages mentioned in https://github.com/hytechimaging/sammo-boat/pull/175#issuecomment-1326274896. So there's something fishy about this PR.

I'm taking a look.

File: src/gui/attribute_table.py [hunk 1]
```diff
@@ -21,6 +21,48 @@ def refresh(table, layerName):
...
+                "glareTo",
+            ]:
+                view.setColumnWidth(
+                    i,
+                    QLabel("")
+                    .fontMetrics()
+                    .boundingRect(view.model().headerData(i, 1))
+                    .width()
+                    + 5,
+                )
```
Comment (pblottiere) [inline src/gui/attribute_table.py:65]: It's definitely this part of the code which causes the segfaults. Can we remove it from the PR for now?

Comment (SebastienPeillet) [reply src/gui/attribute_table.py:65]: All of it or just the `elif` part ?

Comment (pblottiere) [reply src/gui/attribute_table.py:65]: It's actually the `view.setColumnWidth(...)` which causes the segfault, so all of it.

Comment (SebastienPeillet) [reply src/gui/attribute_table.py]: ok it's done.

Just to know, which version of Qt do you use ?

Comment (pblottiere) [reply src/gui/attribute_table.py]: image

**Comment by pblottiere** [comment]
Thanks @SebastienPeillet, it looks good to me :+1:

**Pull request merged**